app/models/post.rb

```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  has_many :likes, foreign_key: 'posts_id'
+  has_many :comments, foreign_key: 'posts_id'
+  belongs_to :author, class_name: 'User', dependent: :destroy, foreign_key: 'author_id'
+
+  def update_posts_couter
```

Comment:
- [OPTIONAL]  You can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the #update_posts_couter method every time a new post has been saved to database, then make the update_posts_couter method private